specifyweb/frontend/js_src/lib/components/Statistics/StatsSpec.tsx

```diff
@@ -0,0 +1,515 @@
+import { getDateInputValue } from '../../utils/dayJs';
+import type { IR } from '../../utils/types';
+import { formatNumber } from '../Atoms/Internationalization';
+import { queryFieldFilters } from '../QueryBuilder/FieldFilter';
+import { statsText } from '../../localization/stats';
+import { formattedEntry } from '../WbPlanView/mappingHelpers';
+import type { BackendStatsResult, StatCategoryReturn } from './types';
+import { userInformation } from '../InitialContext/userInformation';
+import { ensure } from '../../utils/types';
+import { urlSpec } from './definitions';
+
+type StatsSpec =
+  | {
+      readonly [CATEGORY_NAME in keyof BackendStatsResult]: {
+        readonly label: string;
+        readonly categories: (
+          backendStatsResult: BackendStatsResult[CATEGORY_NAME] | undefined
+        ) => StatCategoryReturn;
+      };
+    }
+  | {
+      readonly [CATEGORY_NAME in string]: {
+        readonly label: string;
+        readonly categories: () => StatCategoryReturn;
+      };
+    };
+
+/*
+ * TODO: Return objects directly instead of returning a callback since
+ *  Code now supports relative dates
+ */
+// @ts-expect-error
+export const statsSpec: IR<StatsSpec> = {
+  [statsText.collection()]: {
+    holdings: {
+      label: statsText.holdings(),
+      categories: () => ({
+        specimens: {
+          label: statsText.collectionObjects(),
+          spec: {
+            type: 'QueryBuilderStat',
+            tableName: 'CollectionObject',
+            fields: [
+              {
+                path: formattedEntry,
+                isDisplay: true,
+                operStart: queryFieldFilters.any.id,
+              },
+            ],
+          },
+        },
+        preparations: {
+          label: statsText.preparations(),
+          spec: {
+            type: 'QueryBuilderStat',
+            tableName: 'Preparation',
+            fields: [
+              {
+                path: formattedEntry,
+              },
+              { path: 'countAmt' },
+            ],
+          },
+        },
+        typeSpecimens: {
+          label: statsText.typeSpecimens(),
+          spec: {
+            type: 'QueryBuilderStat',
+            tableName: 'Determination',
+            fields: [
+              {
+                path: formattedEntry,
+              },
+              {
+                path: 'typeStatusName',
+                operStart: queryFieldFilters.equal.id,
+                isNot: true,
+              },
+            ],
+          },
+        } /*
+        familiesRepresented: {
```

Comment:
Yep, I'll be shifting all the node number counting algorithms I had to a different branch - it might be needed in the next year since herb rbge plans to add a million taxons within the next two years